docs/runtime_image.md

```diff
@@ -0,0 +1,83 @@
+
+The following scheme illustrates the build workflow:
+
+![s2i workflow](./runtime-image-flow.png "s2i workflow")
+
+
+## How it works?
+
+To make this work S2I needs to know the following:
+
+* builder image
+* mapping of the artifacts
+* runtime image
+
+And all these information you should specify during S2I invocation:
+
+    s2i build <repo> <builder-image> <my-app> --runtime-image <runtime-image> --runtime-artifact </path/to/artifact>
+
+The only new here are `--runtime-image` and `--runtime-artifact` options. The latter option is used to specify a full path to file (or directory) inside of the builder image and a corresponding directory on the runtime image where S2I will copy these files. Note that you can specify this option multiple times.
+
+For our example S2I will do the following steps:
+
+1. runs builder container and performs a build (as usual)
+1. after build finishes but before stopping runtime container, it downloads artifacts from the builder and places them to a temporary directory on the host
+1. runs container with runtime image
+1. uploads scripts and build artifacts from a temporary directory into `WORKDIR` of a runtime container
+1. runs `assemble-runtime` script in a runtime container
+1. commits container with runtime image
+1. stops build and runtime containers
+
+## Details
+
+### Format and behavior of `--runtime-artifact` option
+
+`--runtime-artifact` option have a format `<source>[:<destination>]`. Here is an example of sample values:
+
+| Value                   | Meaning |
+|-------------------------|---------|
+| `/tmp/app.war`          | `/tmp/app.war` file will be extracted from a builder image and uploaded into the `WORKDIR` of a runtime image |
+| `/tmp/app.war:.`        | the same as above |
+| `/tmp/app.war:app/dist` | `/tmp/app.war` file will be copied from a builder image into into the `app/dist` subdirectory of the `WORKDIR` of a runtime image |
+| `/opt/data`             | `/opt/data` directory will be copied from a builder image into the `WORKDIR` of a runtime image |
+
+Keep in mind that the `source` must be an absolute path. The `destination` must be a relative path to directory, it must exist and path must not start from `..` 
+
+During copying the artifacts, S2I will modify their permissions. All directories and files with executable bit will be uploaded with `0755` mode. Other files will have `0644` mode.
+
+### `assemble-runtime` script requirements
+
+`assemble-runtime` script can be written in any programming language. S2I searches the following locations for this script in the following order:
```

Comment:
yeah, this is certainly not the only place we do it..the reason it caught my eye is because of the "any programming language" comment which feels like a weird thing to say after "script".

maybe "assemble-runtime can be any executable script or binary" would be less confusing?